### DISCUSSION TITLE
icubmod and icubmoddev are NO longer generated

### DISCUSSION THREAD

**Discussion by pattacini** [robotology/community#68] [Announcements]
Hi _icubers_,

Please take a note of this important announcement.

Since https://github.com/robotology/icub-main/commit/ef3a6a68c1918083ef8fc9ee7fae3a84d660209c, **`icubmod`** library so as its related binary **`icubmoddev`** are **NO** longer generated during the compilation of `icub-main`.

Therefore, three consequences are given for code/packages depending on `icub-main`:
1. The dependency from **`icubmod`** in the **`CMakeLists.txt`** file needs to be removed. Such dependency was not required since long, due to the new **dynamic plugins** system.
2. The following C++ directives need to be removed from the code as well:
   - **`YARP_DECLARE_DEVICES(icubmod)`**
   - **`YARP_REGISTER_DEVICES(icubmod)`**
   - **`YARP_DECLARE_PLUGINS(icubmod)`**
   - **`YARP_REGISTER_PLUGINS(icubmod)`**
3. References to **`icubmoddev`** (e.g. in `xml` applications) need to be replaced with references to **`yarpdev`** binary.

See #33 for further details.

**Comment by pattacini** [comment]
As @Tobias-Fischer correctly pointed out, it is also required to remove the following lines from the C++ code:

``` cpp
YARP_DECLARE_DEVICES(icubmod)
YARP_REGISTER_DEVICES(icubmod)
```

**Comment by traversaro** [comment]
@pattacini @mbrunettini can we increase the minor version number of icub-main such that we can cmake depend on the latest version of `icub-main` (after this modification)?

**Comment by traversaro** [comment]
Even the patch version was ok, thanks!

**Comment by pattacini** [comment]
Semantically speaking, it was not a patch but a sort of change in the "_interface_", hence "minor" is more appropriate.

**Comment by traversaro** [comment]
:+1:

**Comment by traversaro** [comment]
I was just scared since the minor version was not updated since 2010 : https://github.com/robotology/icub-main/tags .

**Comment by pattacini** [comment]
Time has come to update it, then :smirk: 
I'm fine with changing the patch too, so whoever admin wants to kick in and do that, there's no problem.

**Comment by jeljaik** [comment]
Bit more details can also be found here: https://github.com/robotology/QA/issues/33#issue-81497604
The original post is missing **`YARP_DECLARE_PLUGINS(icubmod)`** and **`YARP_REGISTER_PLUGINS(icubmod)`**.

**Comment by pattacini** [comment]
Thanks @jeljaik; I've updated the post.
